File: 1911/1911.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAlternatingSum = function(nums) {
    let sumE = 0;
    let sumO = 0;
    for (let i = nums.length-1; i >= 0; i--){
        let tempE = Math.max(sumO + nums[i], sumE);
        let tempO = Math.max(sumE - nums[i], sumO);
        sumE = tempE;
        sumO = tempO;}
    return sumE;};

File: 1911/1911.cpp
class Solution {
public:
    long long maxAlternatingSum(vector<int>& nums) {
        long long sumOdd = 0;
        long long sumEven = 0;

        for(auto it = nums.rbegin(); it != nums.rend(); ++it) {
            long long tempEven = max(sumOdd + *it, sumEven);
            long long tempOdd = max(sumEven - *it, sumOdd);
            sumOdd = tempOdd;
            sumEven = tempEven;
        }

        return sumEven;
    }
};
